### ISSUE TITLE
Add linting rules to block Developer Preview and Experimental features

### ISSUE THREAD

**Issue by marleypowell** [angular-eslint/angular-eslint#2033]
We don't want to accidentally use features that are in Developer Preview or Experimental. There doesn't seem to be a way to stop this other than through manual review.

This isn't something that will be added to the compiler but maybe it's possible to lint them? https://github.com/angular/angular/issues/57656

**Comment by json-derulo** [comment]
This looks like a duplicate of #1721

**Comment by marleypowell** [comment]
> This looks like a duplicate of #1721

yeah, thanks it looks to be a duplicate. although it would be good if the other issue mentioned `experimental` features too.
